projects/collection-editor-library/src/lib/services/editor/editor.service.ts

```diff
@@ -308,25 +308,46 @@ export class EditorService {
   }
 
   async getMaxScore() {
+    let maxScore = 0;
     const rootNode = this.treeService.getFirstChild();
     const metadata = _.get(rootNode, 'data.metadata');
-    const questionIds = this.getContentChildrens();
-    if (metadata.shuffle) {
-      if (metadata.maxQuestions && !_.isEmpty(questionIds) ) {
-        const { questions } =  await this.getQuestionList(_.take(questionIds, metadata.maxQuestions)).toPromise();
-        const maxScore = this.calculateMaxScore(questions);
-        return maxScore;
-      } else {
-        return questionIds.length;
+    let questionIds = [];
+    let questionCount = 0;
+    if (_.get(this.editorConfig, 'config.maxDepth') === 0) {
+      const allQuestionIds = this.getContentChildrens();
+      questionCount = metadata.maxQuestions ? metadata.maxQuestions : allQuestionIds.length;
+      questionIds = _.take(allQuestionIds, questionCount);
+    }
+    if (_.get(this.editorConfig, 'config.maxDepth') === 1) {
```

Comment:
@rajnishdargan What will happen if the `maxDepth` increased in the config?

Reply:
@pallakartheekreddy Modified the method to handle the increase  of  `maxDepth`